internal/dag/builder.go

```diff
@@ -532,7 +537,9 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 				return nil
 			}
 
+			vhost := sw.values["vhost"] // Copy off vhost
 			sw, commit := b.WithObject(delegate)
+			sw.WithValue("vhost", vhost)
```

Comment:
I don't think this is correct. A HTTPProxy can be included by many different roots. I that case we cannot say that its status is related to any specific ghost. Please see https://github.com/projectcontour/contour/pull/1499